modules/exploits/windows/local/cve_2022_26904_superprofile.rb

```diff
@@ -0,0 +1,254 @@
+      target_not_presently_supported
+      return CheckCode::Appears('Vulnerable Windows 10 v1709 build detected!')
+    elsif (build_num_gemversion >= Rex::Version.new('10.0.15063.0')) # Windows 10 v1703
+      target_not_presently_supported
+      return CheckCode::Appears('Vulnerable Windows 10 v1703 build detected!')
+    elsif (build_num_gemversion >= Rex::Version.new('10.0.14393.0')) # Windows 10 v1607 / Windows Server 2016 v1607
+      target_not_presently_supported
+      return CheckCode::Appears('Vulnerable Windows 10 v1607 build detected!')
+    elsif (build_num_gemversion >= Rex::Version.new('10.0.10586.0')) # Windows 10 v1511
+      target_not_presently_supported
+      return CheckCode::Appears('Vulnerable Windows 10 v1511 build detected!')
+    elsif (build_num_gemversion >= Rex::Version.new('10.0.10240.0')) # Windows 10 v1507
+      target_not_presently_supported
+      return CheckCode::Appears('Vulnerable Windows 10 v1507 build detected!')
+    elsif (build_num_gemversion >= Rex::Version.new('6.3.9600.0')) # Windows 8.1/Windows Server 2012 R2
+      target_not_presently_supported
+      return CheckCode::Appears('Vulnerable Windows 8.1/Windows Server 2012 R2 build detected!')
+    elsif (build_num_gemversion >= Rex::Version.new('6.2.9200.0')) # Windows 8/Windows Server 2012
+      target_not_presently_supported
+      return CheckCode::AppearsAppears('Vulnerable Windows 8/Windows Server 2012 build detected!')
+    elsif (build_num_gemversion >= Rex::Version.new('6.1.7601.0')) # Windows 7 SP1/Windows Server 2008 R2 SP1
+      target_not_presently_supported
+      return CheckCode::Appears('Vulnerable Windows 7/Windows Server 2008 R2 build detected!')
+    elsif (build_num_gemversion >= Rex::Version.new('6.1.7600.0')) # Windows 7/Windows Server 2008 R2
+      target_not_presently_supported
+      return CheckCode::Appears('Vulnerable Windows 7/Windows Server 2008 R2 build detected!')
+    elsif (build_num_gemversion >= Rex::Version.new('6.0.6002.0')) # Windows Server 2008 SP2
+      target_not_presently_supported
+      return CheckCode::Appears('Windows Server 2008/Windows Server 2008 SP2 build detected!')
+    else
+      return CheckCode::Safe('The build number of the target machine does not appear to be a vulnerable version!')
+    end
+  end
+
+  def target_not_presently_supported
+    print_warning('This target is not presently supported by this exploit. Support may be added in the future!')
+    print_warning('Attempts to exploit this target with this module WILL NOT WORK!')
+  end
+
+  def check_target_is_running_supported_windows_version
+    if sysinfo['OS'].match('Windows').nil?
+      fail_with(Failure::NotVulnerable, 'Target is not running Windows!')
+    elsif sysinfo['OS'].match('Windows 10').nil? && sysinfo['OS'].match('Windows 11').nil? && sysinfo['OS'].match('Windows Server 2022').nil?
+      fail_with(Failure::BadConfig, 'Target is running Windows, its not a version this module supports! Bailing...')
```

Comment:
```suggestion
      fail_with(Failure::NoTarget, 'Target is running Windows, its not a version this module supports! Bailing...')
```

Should be using [Failure::NoTarget](https://github.com/rapid7/metasploit-framework/wiki/Handling-Module-Failures-with-%60fail_with%60) in this case.

Reply:
Resolved in https://github.com/rapid7/metasploit-framework/pull/16382/commits/f7c271aaf4188adf008bb39abfc7d2838051a059